src/Http/Middleware/DoublePassMiddleware.php

```diff
@@ -45,14 +47,14 @@ public function getCallable(): callable
         return $this->callable;
     }
 
-    protected function decorateHandler(RequestHandlerInterface $handler): callable
+    protected function decorateHandler(RequestHandlerInterface $handler, $response): callable
     {
-        return function ($request, $response) use ($handler) {
-            try {
-                return $handler->handle($request);
-            } catch (Exception $e) {
-                return $response;
+        return function ($request, $res) use ($handler, $response) {
+            if ($res !== $response) {
+                throw new Exception('Callable should not modify response instance before calling $next.');
```

Comment:
could this also be a RuntimeException?

Reply:
This is `Cake\Core\Exception` which is a `RuntimeException` extension.